src/frameworks/utils.ts

```diff
@@ -63,20 +63,47 @@ export async function warnIfCustomBuildScript(
   }
 }
 
-function proxyResponse(original: ServerResponse, next: () => void) {
-  return (response: IncomingMessage | ServerResponse) => {
-    const { statusCode, statusMessage } = response;
-    if (!statusCode) {
-      original.end();
-      return;
-    }
-    if (statusCode === 404) {
-      return next();
-    }
-    const headers = "getHeaders" in response ? response.getHeaders() : response.headers;
-    original.writeHead(statusCode, statusMessage, headers);
-    response.pipe(original);
-  };
+export function proxyResponse(req: IncomingMessage, res: ServerResponse, next: () => void) {
+  const proxiedRes = new ServerResponse(req);
+  const buffer: [string, any[]][] = [];
```

Comment:
Very ingenious. I like this pattern a lot.
Should also make troubleshooting easier in the future.